### PR TITLE
Uplift k8s to 1.20.2 and Fix a small bug in kubelet args

### PR DESCRIPTION
**Pull request by kashifest**
Uplift k8s to 1.20.2 and Fix a small bug in kubelet args

### CODE REVIEW

**Comment by fmuyassarov** [comment]
@maelk

File: vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
         kubeletExtraArgs:
           node-labels: "metal3.io/uuid={{ '{{ ds.meta_data.uuid }}' }}"
           provider-id: "metal3://{{ '{{ ds.meta_data.uuid }}' }}"
-          featureGates: "AllAlpha=false,RunAsGroup=true"
+          feature-gates: "AllAlpha=false,RunAsGroup=true"
```
Comment (furkatgofurov7) [inline vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml:27]: Good catch, how come this was missed in the previous patch? :)

**Review by furkatgofurov7 — COMMENTED** [review]
I suggest running v1alpha3 jobs as well.

**Comment by fmuyassarov** [comment]
@furkatgofurov7 all the CI checks passed now.

**Comment by jan-est** [comment]
LGTM

**Review by jan-est — APPROVED** [review]

**Comment by fmuyassarov** [comment]
approving it since we are sure that this was the bug that was causing the feature tests/upgrade test to fail. And we ran all the CI checks here to double check it.
/approve

**Pull request merged**